feat(android_intent_plus): support for intent as URI

jaumard opened the pull request:
## Description

add support for parsing URI intent into android_intent_plus

## Related Issues

Fix https://github.com/fluttercommunity/plus_plugins/issues/2164

## Checklist

- [x] I read the [Contributor Guide](https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] I titled the PR using [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0).
- [x] I did not modify the `CHANGELOG.md` nor the plugin version in `pubspec.yaml` files.
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate that with a `!` in the title as explained in [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0)).
- [x] No, this is *not* a breaking change.

miquelbeltran commented:
Please follow the contribution guidelines.

1. Don't modify the changelog.
2. Don't modify the pubspec.
3. Provide a description of the changes.

```
'intent://payment#Intent;action=ch.twint.action.TWINT_PAYMENT;scheme=twint;S.code=98815;S.startingOrigin=EXTERNAL_WEB_BROWSER;S.browser_fallback_url=;end';
```

This also looks suspicious, why include an intent to a payments' app with some obscure code.

jaumard commented:
Sorry I read it too fast and miss the NOT in `I did not modify the CHANGELOG.md nor the plugin version in pubspec.yaml files.` I can revert those. 

> This also looks suspicious, why include an intent to a payments' app with some obscure code.

I just took the one I had for my use case, it come from [safer pay](https://docs.saferpay.com) payment provider we uses. This code is internal to TWINT Swiss app usage, they use this code to verify a payment. I don't have other example to put in the example app, I can remove it completely if this is necessary but I needed that to test that it was actually working. 

Why closing the PR instead of asking for changes? Should I bother make another one or just not use this package?

miquelbeltran commented on packages/android_intent_plus/CHANGELOG.md:
Should be reverted

jaumard replied:
Done

miquelbeltran commented on packages/android_intent_plus/pubspec.yaml:
Should be reverted

jaumard replied:
Done

miquelbeltran commented on packages/android_intent_plus/example/lib/main.dart:
```diff
@@ -36,6 +36,8 @@ class MyApp extends StatelessWidget {
 /// Holds the different intent widgets.
 class MyHomePage extends StatelessWidget {
   const MyHomePage({super.key});
+  static const _intent =
+      'intent://payment#Intent;action=ch.twint.action.TWINT_PAYMENT;scheme=twint;S.code=98815;S.startingOrigin=EXTERNAL_WEB_BROWSER;S.browser_fallback_url=;end';
```
Let's replace this with a generic intent URI something like `example://example`.

jaumard replied:
That's not a valid intent URI, intent URI should start with intent, I have used toUri to generate the same alarm intent as the first button so both should do the same now.

miquelbeltran commented on packages/android_intent_plus/example/pubspec.yaml:
Change should be reverted. Use `melos bootstrap` as explained in the contributor guidelines to configure the example.

jaumard replied:
Done

miquelbeltran commented on packages/android_intent_plus/android/src/main/java/dev/fluttercommunity/plus/androidintent/MethodCallHandlerImpl.java:
```diff
@@ -91,7 +92,14 @@ public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) {
+      try {
+        intent = sender.parse(call.argument("uri"));
+        sender.send(intent);
+      } catch (URISyntaxException e) {
+        result.error("WRONG_URI", "Uri's syntax is wrong", null);
```
add the exception message to the `result.error` so users will be able to see it, i.e. the error details: https://api.flutter.dev/javadoc/io/flutter/plugin/common/MethodChannel.Result.html#error(java.lang.String,java.lang.String,java.lang.Object)

jaumard replied:
I have put error code and error message which is enough to get feedback, error details is optional. But anyway I have added the message for both.

miquelbeltran reviewed (changes requested):
Let's keep it in this PR.

I did a quick overview and marked some stuff to be changed and improved. 

I will have to do an actual test on my device before approving.

jaumard commented:
I have reverted pub spec of example project but now the CI is broken saying the method I added doesn't exist. I don't have that issue locally using Melos bootstrap command... not sure what happen

miquelbeltran commented:
No worries, this is another issue that should be solved on the CI side.

I pushed some changes. The happy path works, but I am unable to get a parsing exception no matter what strings I send.

I want to take a look at the native tests, I am not even sure if these run on CI or not, in case we want to add a test here.

jaumard commented:
@miquelbeltran to test the error case I used 'intent:#Intent;;end', with this I was able to see the error in the logs

miquelbeltran reviewed (approved):
I think it looks good now, but I need to fix first the build scripts on `main` branch, so I will come back to merge this once that is fixed.

jaumard commented:
No problem thanks for your help @miquelbeltran !

miquelbeltran commented:
Thanks to you!

miquelbeltran commented:
Thanks again @jaumard for the contribution!

jaumard commented:
Thanks for looking at it quickly, I'll use the git dep until it's on pub. What is the strategy for your release?

miquelbeltran commented:
I think we should prepare a release batch sometime soon, but I need to talk with the other maintainers if there are any blockers
